OpenRA.Game/Traits/Player/Shroud.cs

```diff
@@ -183,7 +182,14 @@ void ITick.Tick(Actor self)
 				var oldResolvedType = resolvedType[puv];
 				resolvedType[puv] = type;
 				if (type != oldResolvedType)
+				{
+					if (type == ShroudCellType.Visible)
+						RevealedCells++;
+					else
+						RevealedCells--;
```

Comment:
Won't this break the count if the type changed from fog to shroud? e.g. if the player picks up a bad crate or is the victim of a spy in the radar dome.

Reply:
changing the `else` to `else if (oldResolvedType == ShroudCellType.Visible)` might be the simplest fix here.